pkg/kubelet/dockershim/naming.go

```diff
@@ -92,7 +102,9 @@ func parseSandboxName(name string) (*runtimeapi.PodSandboxMetadata, error) {
 	name = strings.TrimPrefix(name, "/")
 
 	parts := strings.Split(name, nameDelimiter)
-	if len(parts) != 6 {
+	// Tolerate the random suffix.
+	// TODO(random-liu): Remove 7 field case when docker 1.11 is deprecated.
+	if len(parts) != 6 && len(parts) != 7 {
```

Comment:
I mean that according to the current logic, there is no way that kubelet will create 2 containers with the same name but different suffix. Just like we are sure that kubelet will not create 2 containers with the same attempt (if there is no such name conflict bug).